File: solutions_5648941810974720_1/C++/javra/phone.cpp
#include <cstring>
#include <cmath>
#include <climits>
#include <vector>
#include <deque>
#include <set>
#include <map>
#include <string>
#include <iostream>
#include <algorithm>
#include <functional>
#include <utility>
using namespace std;
using ll = long long;
using ull = unsigned long long;

void solve()
{
    string S;
    cin >> S;

    sort(S.begin(), S.end());
    int count[100] = {0};
    for(char c : S)
        count[c] += 1;

    string phone;
    
    while(count['Z'] > 0)
    {
        phone += '0';
        for(char c : "ZERO")
            count[c] -= 1;
    }
    while(count['W'] > 0)
    {
        phone += '2';
        for(char c : "TWO")
            count[c] -= 1;
    }
    while(count['X'] > 0)
    {
        phone += '6';
        for(char c : "SIX")
            count[c] -= 1;
    }
    while(count['G'] > 0)
    {
        phone += '8';
        for(char c : "EIGHT")
            count[c] -= 1;
    }
    while(count['U'] > 0)
    {
        phone += '4';
        for(char c : "FOUR")
            count[c] -= 1;
    }
    while(count['O'] > 0)
    {
        phone += '1';
        for(char c : "ONE")
            count[c] -= 1;
    }
    while(count['F'] > 0)
    {
        phone += '5';
        for(char c : "FIVE")
            count[c] -= 1;
    }
    while(count['V'] > 0)
    {
        phone += '7';
        for(char c : "SEVEN")
            count[c] -= 1;
    }
    while(count['T'] > 0)
    {
        phone += '3';
        for(char c : "THREE")
            count[c] -= 1;
    }
    while(count['N'] > 0)
    {
        phone += '9';
        for(char c : "NINE")
            count[c] -= 1;
    }

    for(int i = 1; i < 100; i++)
    {
        if(count[i] < 0)
        {
            cerr << "oops " << i << " " << count[i] << "\n";
        }
    }

    sort(phone.begin(), phone.end());
    cout << phone;
    
}

int main()
{
    cout.precision(8);
    int T;
    cin >> T;
    for(int t = 1; t <= T; t++)
    {
        cout << "Case #" << t << ": ";
        solve();
        cout << endl;
    }
    return 0;
}

File: solutions_5648941810974720_0/java/FlyingAnt/Phone.java
import java.util.Scanner;

public class Phone {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        for (int kase = 1, kases = scanner.nextInt(); kase <= kases; kase++) {
            String ss = "";
            while (ss.isEmpty()) {
                ss = scanner.nextLine();
            }
            int[] s = new int[26];
            int[] count = new int[10];

            for (char c : ss.toCharArray()) {
                s[ind(c)]++;
            }

            // ZERO
            while (s[ind('Z')] > 0) {
                count[0]++;
                remove("ZERO", s);
            }

            // SIX
            while (s[ind('X')] > 0) {
                count[6]++;
                remove("SIX", s);
            }

            // EIGHT
            while (s[ind('G')] > 0) {
                count[8]++;
                remove("EIGHT", s);
            }

            // FOUR
            while (s[ind('U')] > 0) {
                count[4]++;
                remove("FOUR", s);
            }

            // THREE
            while (s[ind('H')] > 0) {
                count[3]++;
                remove("THREE", s);
            }

            // TWO
            while (s[ind('W')] > 0) {
                count[2]++;
                remove("TWO", s);
            }

            // ONE
            while (s[ind('O')] > 0) {
                count[1]++;
                remove("ONE", s);
            }

            // FIVE
            while (s[ind('F')] > 0) {
                count[5]++;
                remove("FIVE", s);
            }

            // SEVEN
            while (s[ind('V')] > 0) {
                count[7]++;
                remove("SEVEN", s);
            }

            // NINE
            while (s[ind('E')] > 0) {
                count[9]++;
                remove("NINE", s);
            }

            String ans = "";
            for (int i = 0; i < 10; i++) {
                while (count[i] > 0) {
                    count[i]--;
                    ans += "" + i;
                }
            }
            System.out.println("Case #" + kase + ": " + ans);
        }
    }

    private static void remove(String number, int[] s) {
        for (char c : number.toCharArray()) {
            s[ind(c)]--;
        }
    }

    private static int ind(char c) {
        return c - 'A';
    }
}
